sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamJoinRelBoundedVsBoundedTest.java

```diff
@@ -105,6 +105,34 @@ public void testLeftOuterJoin() throws Exception {
     pipeline.run();
   }
 
+  @Test
+  public void testLeftOuterJoinWithEmptyTuplesOnRightSide() throws Exception {
+    String sql =
+        "SELECT *  "
+            + "FROM ORDER_DETAILS1 o1"
+            + " LEFT OUTER JOIN (SELECT * FROM ORDER_DETAILS2 WHERE FALSE) o2"
```

Comment:
What should happen for other joins (inner) in a similar situation?

Reply:
INNER JOIN should return empty results. I will add a test for it.